File: backend/src/websockets/protocol/responses.py
from backend.src.engine.board import CellState
from backend.src.engine.game import GamePhase, PlayerId
from backend.src.models.ship_status import ShipStatus
from backend.src.websockets.protocol.message_types import Response, ResponseTypes


class CreateGameResponse(Response):
    type: ResponseTypes = ResponseTypes.GAME_CREATED
    code: str


class JoinGameResponse(Response):
    type: ResponseTypes = ResponseTypes.JOINED
    code: str


class GetStateResponse(Response):
    type: ResponseTypes = ResponseTypes.STATE

    phase: GamePhase

    currentPlayer: PlayerId | None
    winner: str | None

    # TODO vraiment utile ?
    # you: PlayerState
    # opponent: PlayerState

    # TODO va surement changer pour un board dto qui contient le array de CellState et d'autres metadata propres au board
    yourBoard: list[list[CellState]]
    enemyBoard: list[list[CellState]]

    ships: list[ShipStatus]


class ErrorResponse(Response):
    type: ResponseTypes = ResponseTypes.ERROR
    message: str

File: frontend/src/protocol/Responses.ts
import type {Response} from "./MessageType.ts";
import type {GamePhases} from "../types/GamePhase.ts";
import type {CellState} from "../types/CellState.ts";
import type {ShipStatus} from "../models/ShipStatus.ts";

export interface CreateGameResponse extends Response {
    code: string;
}

export interface JoinGameResponse extends Response {
    code: string;
}

export interface GetStateResponse extends Response {
    phase: GamePhases;
    currentPlayer?: string;
    winner?: string;
    yourBoard: CellState[][];
    enemyBoard: CellState[][];
    ships: ShipStatus[];
}

export interface ErrorResponse extends Response {
    message: string;
}
